.github/workflows/build.yml

```diff
@@ -151,8 +157,20 @@ jobs:
           fi
           eval "$shell"
 
+      - name: Docker Login
+        env:
+          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
+          DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
+        run: |
+          export DOCKER_CONFIG="$PWD/.docker"
+          mkdir "$DOCKER_CONFIG"
+          trap 'find "$DOCKER_CONFIG" -type f -exec shred -fuvz "{}" + || :; rm -rf "$DOCKER_CONFIG"' EXIT
+          docker login --username "$DOCKER_HUB_USERNAME" --password-stdin <<<"$DOCKER_HUB_PASSWORD"
+          unset DOCKER_HUB_USERNAME DOCKER_HUB_PASSWORD
```

Comment:
I swear I commented about this, but I can't find my comment now :sob:

This won't work as a separate step because the config gets removed as part of it (intentionally), so I think we need instead to revert this part of the changes so this stays in the push step, and I've added new `DOCKER_HUB_SIGNING_USERNAME` and `DOCKER_HUB_SIGNING_PASSWORD` secrets that can be provided to the signing step -- we'll also need to mount `$DOCKER_CONFIG` into the image-signer-verifier container and set `$HOME` appropriately such that the image-signer-verifier can access it.

Reply:
Where does it get removed?

Reply:
```bash
          trap 'find "$DOCKER_CONFIG" -type f -exec shred -fuvz "{}" + || :; rm -rf "$DOCKER_CONFIG"' EXIT
```

This line deletes the entire `$DOCKER_CONFIG` directory (including using `shred` on the contents, for good measure) the minute the shell exits, which is at the end of this `run:` block.

Reply:
aaaaah! I didn't know that's what it does. I thought it ran first